proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -0,0 +1,717 @@
+from a DXContainer.
+
+Other requirements for the overall effort that need to influence the design:
+1) Enable other frontends to utilize the DirectX backend.
+2) Plan for the evolution of root signatures in future versions.
+
+## Background
+
+What is root signature
+https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signatures
+
+How to create root signature in hlsl
+https://learn.microsoft.com/en-us/windows/win32/direct3d12/specifying-root-signatures-in-hlsl
+
+Local/Global root signature
+https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-state-object
+
+There're different root signature in hlsl.
+
+With RootSignature attribute.
+
+```
+
+    #define RS "RootFlags( ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT | " \
+              "DENY_VERTEX_SHADER_ROOT_ACCESS), " \
+              "CBV(b0, space = 1, flags = DATA_STATIC), " \
+              "SRV(t0), " \
+              "UAV(u0), " \
+              "DescriptorTable( CBV(b1), " \
+              "                 SRV(t1, numDescriptors = 8, " \
+              "                     flags = DESCRIPTORS_VOLATILE), " \
+              "                 UAV(u1, numDescriptors = unbounded, " \
+              "                     flags = DESCRIPTORS_VOLATILE)), " \
+              "DescriptorTable(Sampler(s0, space=1, numDescriptors = 4)), " \
+              "RootConstants(num32BitConstants=3, b10), " \
+              "StaticSampler(s1)," \
+              "StaticSampler(s2, " \
+              "              addressU = TEXTURE_ADDRESS_CLAMP, " \
+              "              filter = FILTER_MIN_MAG_MIP_LINEAR )"
+
+    [RootSignature(RS)]
+    float4 main(float4 coord : COORD) : SV_Target
+    {
+    …
+    }
+
+```
+
+The other mechanism is to create a standalone root signature blob, perhaps to 
+reuse it with a large set of shaders, saving space. There're shader model 
+rootsig_1_0 androotsig_1_1 to generate the standalone root signature blob.
+The name of the define string is specified via the usual /E argument.
+
+For example:
+
+```
+
+dxc.exe /T rootsig_1_1 MyRS1.hlsl /E RS /Fo MyRS1.fxo
+```
+Note that the root signature string define can also be passed on the command 
+line, e.g, /D MyRS1=”…”.
+
+
+In DXR, there're local and global root signature which could be also create 
+root signature in hlsl
+
+```
+
+GlobalRootSignature MyGlobalRootSignature =
+{
+    "DescriptorTable(UAV(u0)),"                     // Output texture
+    "SRV(t0),"                                      // Acceleration structure
+    "CBV(b0),"                                      // Scene constants
+    "DescriptorTable(SRV(t1, numDescriptors = 2))"  // Static index and vertex buffers.
+};
+
+LocalRootSignature MyLocalRootSignature = 
+{
+    "RootConstants(num32BitConstants = 4, b1)"  // Cube constants 
+};
+
+```
+
+There's -setrootsignature option in DXC which attach root signature part to an existing input DXIL container. It is a DXIL container level operation. Not AST or llvm IR.
```

Comment:
nit: This line exceeds the 80 col limit.

Reply:
Fixed.